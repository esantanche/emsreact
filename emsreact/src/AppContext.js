// This is the application context
// We populate it in App.js and provide it to the entire application

import React from "react";

const AppContext = React.createContext(null);

export default AppContext;