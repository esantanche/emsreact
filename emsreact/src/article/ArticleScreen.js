import React, { Component } from 'react';

// Material UI components
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

// Material UI icons
import Clear from '@material-ui/icons/Clear';
import IconButton from '@material-ui/core/IconButton';

// From components library
import FrontGridPane from '../components_library/panes/FrontGridPane';

// The injector allows this component to use services, see README.md in folder 'services'
import { injector } from 'react-services-injector';

/**
 * Screen showing a single article. The text can contain html tags. In this case we render them.
 */
class ArticleScreen extends Component {

    constructor(props) {
        super(props);

        this.state = { article: [ { } ] };

    }

    /**
     * When the component mounts, we fetch the article we have to display.
     *
     * The url that triggers displaying this Screen is like this:
     * /article/:articlenid
     *
     * For example it can be: /article/27
     *
     * The number is the node id of the article in the Drupal backend.
     *
     * This component uses the service ArticleService for it to fetch
     * that specific article.
     */
    componentDidMount() {

        if (!this.props.match.params.articlenid) {

            console.error("ArticleScreen::componentDidMount");
            console.error("this.props.match.params.articlenid is not defined");
            console.error("This is a bug");

            return;
        }

        const { ArticleService } = this.services;

        this.ArticleService = ArticleService;

        var self = this;

        this.ArticleService.fetch_articles({ nid: this.props.match.params.articlenid },
            function (article) {

                self.setState({ article: article });

            }

        )

    }

    render() {

        return (

            <React.Fragment>

                <FrontGridPane>

                    <Card>

                        <CardHeader title={this.state.article[0].title}
                                    action={
                                        <IconButton href="/">
                                            <Clear />
                                        </IconButton>
                                    }/>

                        <CardContent>

                            <Typography variant="body1" component="div">

                                <div dangerouslySetInnerHTML={{ __html: this.state.article[0].body }} />

                            </Typography>

                        </CardContent>

                    </Card>

                </FrontGridPane>

            </React.Fragment>

        )

    }

}

export default injector.connect(ArticleScreen);
