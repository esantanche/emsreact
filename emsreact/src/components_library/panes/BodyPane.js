import React, { Component } from 'react';
import { withStyles } from "@material-ui/core/styles/index";
import PropTypes from 'prop-types';

const styles = {
    bodyPane: {
        backgroundColor: "#FFFFFF",
        overflowY: "hidden"
    },
};

class BodyPane extends Component {

    static propTypes = {
        classes: PropTypes.object.isRequired
    };

    render() {

        const { classes } = this.props;

        return (

            <div className={classes.bodyPane}>

                {this.props.children}

            </div>

        )

    }

}

export default withStyles(styles)(BodyPane);
