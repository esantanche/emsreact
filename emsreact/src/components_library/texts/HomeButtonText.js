import React, { Component } from 'react';
import PropTypes from 'prop-types';

// Material UI components
import Typography from "@material-ui/core/Typography";

// Material UI functions that make classes and width available
import { withStyles } from "@material-ui/core/styles/index";
import withWidth from "@material-ui/core/withWidth";

const styles = {
    HomeButtonText: {
        lineHeight: "1.5em"
    }
};

/**
 * The text component used for the home button.
 * It changes the variant of the Typography component according to
 * device width.
 *
 * On small devices a smaller font is used.
 *
 * Props
 * @param {object} classes Classes injected by withStyles function
 * @param {string} width Material UI device width (xs, sm, md, lg, xl). The function withWidth
 * provides it.
 */
class HomeButtonText extends Component {

    static propTypes = {
        classes: PropTypes.object.isRequired,
        width: PropTypes.string
    };

    render() {

        const { classes, width } = this.props;

        return (

            <Typography variant={ ['xs', 'sm'].indexOf(width) >= 0 ? "body2" : "headline" }
                        className={classes.HomeButtonText}>
                {this.props.children}
            </Typography>

        )

    }

}

export default withWidth()(withStyles(styles)(HomeButtonText));
