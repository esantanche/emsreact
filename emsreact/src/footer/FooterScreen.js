import React, { Component } from 'react';

// Material UI components
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

import withWidth from "@material-ui/core/withWidth/index";
import {withStyles} from "@material-ui/core/styles/index";
import FrontGridPane from '../components_library/panes/FrontGridPane';
import ColouredCard from '../components_library/cards/ColouredCard';
import SeparatorPane from '../components_library/panes/SeparatorPane';

class FooterScreen extends Component {

    render() {

        // const { classes } = this.props;

        const { width } = this.props;

        return (

            <React.Fragment>

                <FrontGridPane>

                    {/*FIXME it may be better to make a separate component for the contact me form*/}

                    <ColouredCard>

                        <CardContent>

                            <Typography variant="title" >

                                contact me form

                            </Typography>

                        </CardContent>

                    </ColouredCard>

                    <SeparatorPane />

                    <ColouredCard>

                        <CardContent>

                            <Typography variant="title" >

                                small print

                            </Typography>

                        </CardContent>

                    </ColouredCard>

                </FrontGridPane>

            </React.Fragment>

        )
    }

}

// FIXME no styles here, move them to component library
// withWidth shouldn't be used as well

export default withWidth()(FooterScreen);


