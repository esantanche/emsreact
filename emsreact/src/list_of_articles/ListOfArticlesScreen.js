import React, { Component } from 'react';
import PropTypes from 'prop-types';

// Material UI components
import Card from '@material-ui/core/Card';

// From components library
import InnerGridPane from '../components_library/panes/InnerGridPane';
import ArticleCard from '../components_library/cards/ArticleCard';

import AppContext from "../AppContext";

/**
 * This Screen shows a list of articles. It's supposed to use the service ArticleService
 * to retrieve them from the backend.
 *
 * If the prop sticky is true, only the articles that are flagged as sticky on the backend
 * will be retrieved.
 *
 * @param {string} topic The topic of articles to show. Like "How I work", "The tools I use",
 * "Success stories", etc.
 * @param {boolean} sticky True if only sticky articles should be displayed
 */
class ListOfArticlesScreen extends Component {

    constructor() {
        super();

        this.state = { articles: [] }

    }

    static propTypes = {
        topic: PropTypes.string,
        sticky: PropTypes.bool
    };

    componentDidMount() {

        // Fetching articles when the component mounts

        const { ArticleService } = this.props.services;

        this.ArticleService = ArticleService;

        var self = this;

        this.ArticleService.fetch_articles({ topic: this.props.topic, sticky: this.props.sticky },
            function (articles) {

                self.setState({ articles: articles });

            }
        )
    }

    componentDidUpdate(prevProps) {

        var self = this;

        // When the topic changes, the props to this component change as well
        // We need to fetch the articles related to the new topic

        // Same thing if the prop 'sticky' changes. Again we have to fetch articles again

        if (prevProps.topic !== this.props.topic ||
            prevProps.sticky !== this.props.sticky) {

            this.ArticleService.fetch_articles({ topic: this.props.topic, sticky: this.props.sticky },
                function (articles) {

                    self.setState({ articles: articles });

                }
            )

        }

    }

    render() {

        // We get articles in an array where each item contains one article
        // We produce another array where each item contains two articles
        // This makes it easier to use the component InnerGridPane to build the list.
        // That component shows two articles at a time (a leftComponent and a
        // rightComponent)

        const articles_pairs = this.state.articles.reduce((articles_pairs, article, index, articles) => {

            if (index % 2 === 0)
                articles_pairs.push(articles.slice(index, index + 2));

            return articles_pairs;

        }, []);

        return articles_pairs.map((pair_of_articles, index) => {

            return (

                <InnerGridPane key={index}

                    leftComponent={ (

                            <ArticleCard image={pair_of_articles[0].field_image}
                                         title={pair_of_articles[0].title}
                                         text_content={pair_of_articles[0].body}
                                         article_node_id={Number(pair_of_articles[0].nid)}>

                            </ArticleCard>

                    ) }

                    rightComponent={ pair_of_articles.length === 2 ? (

                            <ArticleCard image={pair_of_articles[1].field_image}
                                         title={pair_of_articles[1].title}
                                         text_content={pair_of_articles[1].body}
                                         article_node_id={Number(pair_of_articles[1].nid)}>

                            </ArticleCard>

                    ) : (

                        <Card></Card>

                    )}

                />

            )

        })

    }

}

export default props => (
    <AppContext.Consumer>
        {services => <ListOfArticlesScreen {...props} services={services} />}
    </AppContext.Consumer>
);

// export default injector.connect(ListOfArticlesScreen);
