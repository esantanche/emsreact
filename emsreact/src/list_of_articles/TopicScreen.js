import React, { Component } from 'react';

// Material UI components
import Card from '@material-ui/core/Card';
import Avatar from '@material-ui/core/Avatar';
import IconButton from '@material-ui/core/IconButton';
import CardHeader from '@material-ui/core/CardHeader';

// Material UI icons
import Clear from '@material-ui/icons/Clear';

// From components library
import FrontGridPane from '../components_library/panes/FrontGridPane';
import SeparatorPane from "../components_library/panes/SeparatorPane";

// Screen that shows a list of articles
import ListOfArticlesScreen from '../list_of_articles/ListOfArticlesScreen';

// The injector allows this component to use services, see README.md in folder 'services'
//import { injector } from 'react-services-injector';

import AppContext from '../AppContext';


/**
 * This screen shows a list of articles for a given topic.
 *
 * Both sticky and non-sticky articles are shown.
 *
 * For now there is no pagination. All articles are shown.
 *
 * When this screen is mounted, the url is like this:
 * "/articles/:topic"
 *
 * Props will provide the parameter 'topic' in this variable: this.props.match.params.topic
 */
class TopicScreen extends Component {

    constructor(props) {
        super(props);

        this.state = { topic_details: {} };
    }

    componentDidMount() {

        if (!this.props.match.params.topic) {

            console.error("TopicScreen::componentDidMount");
            console.error("this.props.match.params.topic is not defined");
            console.error("This is a bug");

            return;
        }

        const { TopicService } = this.props.services;

        this.TopicService = TopicService;

        var self = this;

        // First time the component is mounted we find the topic we have to pass to
        // the component ListOfArticlesScreen for it to call the service that will
        // return the articles

        // this.props.match.params.topic is the topic in the format that goes on the url
        // this.state.topic_details.name is the same topic in the format to be used to fetch
        // articles
        // For example:
        // how-i-work is good for urls
        // "How I work" is to be used to fetch articles
        // The service TopicService provides topics good to fetch articles from topics
        // good for urls

        // Consider that the object topic_details returned by get_topic_from_url is an object
        // containing many properties, not just the name of the topic as it is needed
        // to fetch articles

        this.TopicService.get_topic_from_url(this.props.match.params.topic, function(topic_details) {

            self.setState({ topic_details: topic_details });

        });

    }

    componentDidUpdate(prevProps) {

        var self = this;

        // When the topic changes, the props to this component change as well
        // We need to transform again the topic as provided by the url into
        // the topic as needed to fetch articles

        if (prevProps.match.params.topic !== this.props.match.params.topic) {

            if (this.TopicService) {
                this.TopicService.get_topic_from_url(this.props.match.params.topic, function(topic_details) {

                    self.setState({ topic_details: topic_details });

                });
            }

        }

    }

    render() {

        const topic_details = this.state.topic_details;

        // If topic_details.name is not defined, it means that the topic_details object that the service TopicService
        // should provide is not yet available.
        // In this case we just render an empty element

        return (

            topic_details.name ?

            <FrontGridPane>

                <Card>
                    <CardHeader
                        avatar={
                            <Avatar aria-label={topic_details.name} style={{ backgroundColor: topic_details.color_for_avatar }}>
                                {topic_details.letter_for_avatar}
                            </Avatar>
                        }
                        action={
                            <IconButton href="/">
                                <Clear />
                            </IconButton>
                        }
                        title={topic_details.name}
                        subheader={topic_details.subheader}
                    />

                    <ListOfArticlesScreen topic={topic_details.name} sticky={false}/>

                    <SeparatorPane/>

                </Card>

            </FrontGridPane>

                :

            <FrontGridPane>

            </FrontGridPane>

        )

    }

}


export default props => (
    <AppContext.Consumer>
        {services => <TopicScreen {...props} services={services} />}
    </AppContext.Consumer>
);

// export default injector.connect(TopicScreen);

