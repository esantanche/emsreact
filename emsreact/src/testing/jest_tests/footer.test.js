// FIXME docs? anything to do?
// Testing everything in folder article
// fixme fix docs


import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

import ListOfArticlesScreen from "../../list_of_articles/ListOfArticlesScreen";
import TopicScreen from "../../list_of_articles/TopicScreen";

import ArticleService from "../../services/article/ArticleService";
import TopicService from "../../services/topic/TopicService";

import AppContext from "../../AppContext";
import ArticleScreen from "../../article/ArticleScreen";
import FooterScreen from "../../footer/FooterScreen";
import BodyPane from "../../components_library/panes/BodyPane";
import HomeButtonText from "../../components_library/texts/HomeButtonText";
import PickMyBrainMessageService from "../../services/pick_my_brain_message/PickMyBrainMessageService";

//const AppArticleService = new ArticleService();
// const AppTopicService = new TopicService();

// const Services = { TopicService: null,
//     ArticleService: AppArticleService,
//     PickMyBrainMessageService: null };


// FIXME this is for the footer


//                            <FooterScreen />

// const json_response_rest_api_call = [
//     {
//         "nid": "58",
//         "title": "Privacy Policy",
//         "body": "This is the body of the privacy policy. It\' for testing",
//         "langcode": "English",
//         "field_ems_topic": "Small print",
//         "sticky": "Off",
//         "field_image": ""
//     }
// ];

// const article_screen_to_be_tested = ( <AppContext.Provider value={Services}>
//                                          <ArticleScreen match={{ params: { articlenid: 58 } }}
//                                          />
//                                       </AppContext.Provider> );

describe('Testing FooterScreen', function() {


    const AppPickMyBrainMessageService = new PickMyBrainMessageService();

    const Services = { TopicService: null,
        ArticleService: null,
        PickMyBrainMessageService: AppPickMyBrainMessageService };

    // const Services = { TopicService: AppTopicService,
    //     ArticleService: AppArticleService,
    //     PickMyBrainMessageService: null };

    const footer_screen_to_be_tested = ( <AppContext.Provider value={Services}>
                                            <FooterScreen />
                                        </AppContext.Provider> );

    // const footer_screen_to_be_tested = (  );

    // const message_dialog_confirmation_message_to_be_tested = ( <MessageDialog open={true} it_is_an_error_message={false}
    //                                                                           title="Message sent"
    //                                                                           message="Thank you for your message!"/> );

    it('FooterScreen renders correctly', () => {

        // const tree = shallow(header_menu_to_be_tested);
        //
        // expect(EnzymeToJson(tree)).toMatchSnapshot();
        const tree = renderer
            .create(footer_screen_to_be_tested)
            .toJSON();
        expect(tree).toMatchSnapshot();

    });

});
