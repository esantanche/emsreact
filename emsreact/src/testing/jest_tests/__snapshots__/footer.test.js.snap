// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FooterScreen FooterScreen renders correctly 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-16-23"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-md-1-57 MuiGrid-grid-lg-2-72"
  />
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-40 MuiGrid-grid-md-10-66 MuiGrid-grid-lg-8-78"
  >
    <div
      className="MuiPaper-root-100 MuiPaper-elevation1-103 MuiPaper-rounded-101 MuiCard-root-99 ColouredCard-light_coloured_card-98"
    >
      <div
        className="MuiCardHeader-root-127"
      >
        <div
          className="MuiCardHeader-content-130"
        >
          <span
            className="MuiTypography-root-133 MuiTypography-headline-138 MuiCardHeader-title-131"
          >
            Pick My Brain!
          </span>
          <span
            className="MuiTypography-root-133 MuiTypography-body1-142 MuiTypography-colorTextSecondary-155 MuiCardHeader-subheader-132"
          >
            Or just ask me a question
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root-157"
      >
        <div
          className="MuiGrid-container-1 MuiGrid-spacing-xs-32-25"
        >
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-40 MuiGrid-grid-sm-6-48"
          >
            <div
              aria-describedby={undefined}
              className="MuiFormControl-root-158 MuiFormControl-marginDense-160 MuiFormControl-fullWidth-161"
            >
              <label
                className="MuiFormLabel-root-167 MuiInputLabel-root-162 MuiInputLabel-formControl-163 MuiInputLabel-animated-166 MuiInputLabel-shrink-165 MuiInputLabel-marginDense-164"
                data-shrink={true}
                htmlFor="email"
              >
                Email
                <span
                  className="MuiFormLabel-asterisk-171"
                >
                   *
                </span>
              </label>
              <div
                className="MuiInput-root-172 MuiInput-fullWidth-179 MuiInput-formControl-173 MuiInput-underline-176"
              >
                <div
                  className="MuiInputAdornment-root-185 MuiInputAdornment-positionStart-186"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-188"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <g>
                      <path
                        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                      />
                    </g>
                  </svg>
                </div>
                <input
                  aria-invalid={false}
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="MuiInput-input-180 MuiInput-inputMarginDense-181"
                  defaultValue={undefined}
                  disabled={false}
                  id="email"
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={undefined}
                  onKeyUp={undefined}
                  placeholder={undefined}
                  readOnly={undefined}
                  required={true}
                  rows={undefined}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-40 MuiGrid-grid-sm-6-48"
          >
            <div
              aria-describedby={undefined}
              className="MuiFormControl-root-158 MuiFormControl-marginDense-160 MuiFormControl-fullWidth-161"
            >
              <label
                className="MuiFormLabel-root-167 MuiInputLabel-root-162 MuiInputLabel-formControl-163 MuiInputLabel-animated-166 MuiInputLabel-shrink-165 MuiInputLabel-marginDense-164"
                data-shrink={true}
                htmlFor="name"
              >
                Name
                <span
                  className="MuiFormLabel-asterisk-171"
                >
                   *
                </span>
              </label>
              <div
                className="MuiInput-root-172 MuiInput-fullWidth-179 MuiInput-formControl-173 MuiInput-underline-176"
              >
                <div
                  className="MuiInputAdornment-root-185 MuiInputAdornment-positionStart-186"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-188"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <g>
                      <path
                        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                      />
                    </g>
                  </svg>
                </div>
                <input
                  aria-invalid={false}
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="MuiInput-input-180 MuiInput-inputMarginDense-181"
                  defaultValue={undefined}
                  disabled={false}
                  id="name"
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={undefined}
                  onKeyUp={undefined}
                  placeholder={undefined}
                  readOnly={undefined}
                  required={true}
                  rows={undefined}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-container-1 MuiGrid-spacing-xs-32-25"
        >
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-40"
          >
            <div
              aria-describedby={undefined}
              className="MuiFormControl-root-158 MuiFormControl-marginDense-160 MuiFormControl-fullWidth-161"
            >
              <label
                className="MuiFormLabel-root-167 MuiInputLabel-root-162 MuiInputLabel-formControl-163 MuiInputLabel-animated-166 MuiInputLabel-shrink-165 MuiInputLabel-marginDense-164"
                data-shrink={true}
                htmlFor="message"
              >
                Message
                <span
                  className="MuiFormLabel-asterisk-171"
                >
                   *
                </span>
              </label>
              <div
                className="MuiInput-root-172 MuiInput-fullWidth-179 MuiInput-formControl-173 MuiInput-multiline-178 MuiInput-underline-176"
              >
                <div
                  className="MuiInputAdornment-root-185 MuiInputAdornment-positionStart-186"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-188"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <g>
                      <path
                        d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-2 12H6v-2h12v2zm0-3H6V9h12v2zm0-3H6V6h12v2z"
                      />
                    </g>
                  </svg>
                </div>
                <div
                  className="Textarea-root-195"
                  style={
                    Object {
                      "height": 19,
                    }
                  }
                >
                  <textarea
                    aria-hidden="true"
                    className="Textarea-shadow-197 Textarea-textarea-196"
                    readOnly={true}
                    rows="1"
                    tabIndex={-1}
                    value=""
                  />
                  <textarea
                    aria-hidden="true"
                    className="Textarea-shadow-197 Textarea-textarea-196"
                    defaultValue={undefined}
                    readOnly={true}
                    rows={1}
                    tabIndex={-1}
                    value=""
                  />
                  <textarea
                    aria-invalid={false}
                    autoComplete={undefined}
                    autoFocus={undefined}
                    className="Textarea-textarea-196 MuiInput-input-180 MuiInput-inputMultiline-182 MuiInput-inputMarginDense-181"
                    defaultValue={undefined}
                    disabled={false}
                    id="message"
                    name={undefined}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={undefined}
                    onKeyUp={undefined}
                    placeholder={undefined}
                    readOnly={undefined}
                    required={true}
                    rows={1}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-container-1 MuiGrid-spacing-xs-32-25"
        >
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-40"
          >
            <button
              className="MuiButtonBase-root-222 MuiButtonBase-disabled-223 MuiButton-root-198 MuiButton-text-200 MuiButton-flat-203 MuiButton-sizeSmall-219 MuiButton-disabled-216"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="-1"
              type="button"
            >
              <span
                className="MuiButton-label-199"
              >
                Submit
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="SeparatorPane-separatorPane-225"
    >
       
    </div>
    <div
      className="MuiGrid-container-1 MuiGrid-spacing-xs-16-23"
    >
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-6-34 MuiGrid-grid-sm-3-45 MuiGrid-grid-md-3-59 MuiGrid-grid-lg-3-73"
      >
        <div
          className="MuiPaper-root-100 MuiPaper-elevation1-103 MuiPaper-rounded-101 MuiCard-root-99"
        >
          <a
            className="MuiButtonBase-root-222 SocialCard-cardButton-227"
            href="https://github.com/esantanche"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
            target="_blank"
          >
            <div
              className="MuiCardMedia-root-228 SocialCard-image_responsive_size-226"
              src={undefined}
              style={
                Object {
                  "backgroundImage": "url(\\"GitHub_Logo.png\\")",
                }
              }
              title=""
            />
            <span
              className="MuiTouchRipple-root-230"
            />
          </a>
        </div>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-6-34 MuiGrid-grid-sm-3-45 MuiGrid-grid-md-3-59 MuiGrid-grid-lg-3-73"
      >
        <div
          className="MuiPaper-root-100 MuiPaper-elevation1-103 MuiPaper-rounded-101 MuiCard-root-99"
        >
          <a
            className="MuiButtonBase-root-222 SocialCard-cardButton-227"
            href="https://medium.com/@esantanche"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
            target="_blank"
          >
            <div
              className="MuiCardMedia-root-228 SocialCard-image_responsive_size-226"
              src={undefined}
              style={
                Object {
                  "backgroundImage": "url(\\"Medium_logo.png\\")",
                }
              }
              title=""
            />
            <span
              className="MuiTouchRipple-root-230"
            />
          </a>
        </div>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-6-34 MuiGrid-grid-sm-3-45 MuiGrid-grid-md-3-59 MuiGrid-grid-lg-3-73"
      >
        <div
          className="MuiPaper-root-100 MuiPaper-elevation1-103 MuiPaper-rounded-101 MuiCard-root-99"
        >
          <a
            className="MuiButtonBase-root-222 SocialCard-cardButton-227"
            href="https://www.linkedin.com/in/emanuelesantanche/"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
            target="_blank"
          >
            <div
              className="MuiCardMedia-root-228 SocialCard-image_responsive_size-226"
              src={undefined}
              style={
                Object {
                  "backgroundImage": "url(\\"LinkedIn_logo.png\\")",
                }
              }
              title=""
            />
            <span
              className="MuiTouchRipple-root-230"
            />
          </a>
        </div>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-6-34 MuiGrid-grid-sm-3-45 MuiGrid-grid-md-3-59 MuiGrid-grid-lg-3-73"
      >
        <div
          className="MuiPaper-root-100 MuiPaper-elevation1-103 MuiPaper-rounded-101 MuiCard-root-99"
        >
          <a
            className="MuiButtonBase-root-222 SocialCard-cardButton-227"
            href="https://stackoverflow.com/users/6171615/emanuele-santanche"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
            target="_blank"
          >
            <div
              className="MuiCardMedia-root-228 SocialCard-image_responsive_size-226"
              src={undefined}
              style={
                Object {
                  "backgroundImage": "url(\\"StackOverflow_logo.png\\")",
                }
              }
              title=""
            />
            <span
              className="MuiTouchRipple-root-230"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      className="SeparatorPane-separatorPane-225"
    >
       
    </div>
    <div
      className="MuiPaper-root-100 MuiPaper-elevation1-103 MuiPaper-rounded-101 MuiCard-root-99 ColouredCard-coloured_card-97"
    >
      <div
        className="MuiCardContent-root-157"
      >
        <div
          className="MuiGrid-container-1 MuiGrid-spacing-xs-8-22 BottomGridPane-container-237"
        >
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-9-37"
          >
            <h2
              className="MuiTypography-root-133 MuiTypography-title-139"
            >
              Emanuele Santanché
            </h2>
            <h3
              className="MuiTypography-root-133 MuiTypography-subheading-140"
            >
              Paris, France and, remotely, worldwide
            </h3>
          </div>
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-3-31 BottomGridPane-right_element-238"
          >
            <a
              className="MuiButtonBase-root-222"
              href="/article/58/privacy-policy"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex="0"
            >
              <h2
                className="MuiTypography-root-133 MuiTypography-title-139"
              >
                Privacy policy
              </h2>
              <span
                className="MuiTouchRipple-root-230"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-md-1-57 MuiGrid-grid-lg-2-72"
  />
</div>
`;
